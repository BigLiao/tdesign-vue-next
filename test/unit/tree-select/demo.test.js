import { mount } from '@vue/test-utils';
import base from '@/examples/tree-select/demos/base.vue';
import multiple from '@/examples/tree-select/demos/multiple.vue';
import filterable from '@/examples/tree-select/demos/filterable.vue';
import lazy from '@/examples/tree-select/demos/lazy.vue';
import prefix from '@/examples/tree-select/demos/prefix.vue';
import collapsed from '@/examples/tree-select/demos/collapsed.vue';

// unit test for component in examples.
describe('TreeSelect', () => {
  it('base demo works fine', () => {
    const wrapper = mount(base);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('multiple demo works fine', () => {
    const wrapper = mount(multiple);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('filterable demo works fine', () => {
    const wrapper = mount(filterable);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('lazy demo works fine', () => {
    const wrapper = mount(lazy);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('prefix demo works fine', () => {
    const wrapper = mount(prefix);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('collapsed demo works fine', () => {
    const wrapper = mount(collapsed);
    expect(wrapper.element).toMatchSnapshot();
  });
});
