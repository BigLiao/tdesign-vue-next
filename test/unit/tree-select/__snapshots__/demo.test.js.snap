// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tree-select base demo works fine 1`] = `
<div
  class="tdesign-tree-select-base"
>
  <t-tree-select
    clearable=""
    data="[object Object],[object Object]"
    modelvalue=""
    placeholder="请选择"
  />
</div>
`;
